### ISSUE TITLE
[Metricbeat] AWS Billing Module accept pairs of group_by values

### ISSUE THREAD

**Issue by thomascate** [elastic/beats#34193]
Currently the AWS billing module takes in an arbitrary number of group_by dimensions. However it uses each one individually pairing it with a group_by tag. Not having multi dimension search makes the data for consolidated billing much less useful.

For example,
If you have 10 accounts and you scan your management account with LINKED_ACCOUNT and SERVICES dimensions.
With the current AWS Billing Module, you'll get two sets of documents.
1. Documents with a summary of total service usage across all accounts.
2. Documents with a summary of total bill for each account.
This means that you cannot drill into how much usage each account had by service.

Ideally the module would accept pairs of group_by values. If you pass those same two group_by dimensions to the api together you get one document per account per service, which allows much deeper filtering.

There are some more details [here](https://discuss.elastic.co/t/metricbeat-aws-billing-module-does-not-combine-group-by-options/322497).

**Comment by BenB196** [comment]
Definitely agree that there is room for improvement around the AWS billing module, recently started poking around at it, and noticed the same issues when trying to analyze the various group_by dimensions. I think part of this might be a limitation of the AWS Billing [API](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetCostAndUsage.html#API_GetCostAndUsage_RequestSyntax) where it only allows for 2 `group_by` dimensions per request.

The interesting thing I found, if you set a standard group by dimension (ex: LINKED_ACCOUNT) and then a tag group by dimension, the module will have both group_bys on the document. The one issue with relying on the tag groups, is that tags aren't guaranteed by AWS to be available in on billing events, and even then it puts more onus on the user to set and use billing tags.

**Comment by BenB196** [comment]
One thing I did consider was potentially use Elasticsearch [transforms](https://www.elastic.co/guide/en/elasticsearch/reference/current/transforms.html) to repivot this data into something more usable, but haven't gotten around to experimenting with it yet.

**Comment by thomascate** [comment]
Yeah, I think defaulting to having the tag group_by is what causes the issue.

I don't think you can pivot the data unless you query it with multiple dimensions in the first place. Since that's the only way to get details such as "Total ec2 usage for account foo" as that requires two dimensions to get that specific data back.

I think letting the user give a pair of group_by options of either type is the most flexible. Then they can do 1 dimension, 1 tag like now, or 2 dimensions, or even 2 tags. Really nice would be a list of pairs, and then we just iterate through each pair.

**Comment by Gawne** [comment]
Hello we would also value the ability to search on multiple group_by dimensions in order to retrieve more granular info on service usages per account when pulling data from a consolidated management account.

We have raised feature request internally with Elastic.

Thanks,
Matt
